lua/starfall/libs_sh/game.lua

```diff
@@ -134,6 +146,83 @@ if SERVER then
 		util.BlastDamage(instance.entity, instance.player, vunwrap(damageOrigin), math.Clamp(damageRadius, 0, 1500), damage)
 	end
 
+	--- Fires a bullet. Bullet made with this function will not have any tracer, you will have to make them yourself.
+	-- @server
+	-- @param number? damage The damage dealt by the bullet. (1-100)
+	-- @param number? force The force of the bullets. (0-100)
+	-- @param number? distance Maximum distance the bullet can travel.
+	-- @param number? hullSize The hull size of the bullet. (0-10)
+	-- @param number? num The amount of bullets to fire. (1-5)
+	-- @param Vector Dir The fire direction.
+	-- @param Vector Spread The spread, only x and y are needed.
+	-- @param Vector src The position to fire the bullets from.
+	-- @param Entity? ignoreEntity The entity that the bullet will ignore when it will be shot.
+	-- @param function? callback Function to be called with attacker, traceResult after the bullet was fired but before the damage is applied (the callback is called even if no damage is applied).
+	function game_library.bulletDamage(damage, force, distance, hullSize, num, dir, spread, src, ignoreEntity, cb)
+		local ent = instance.entity
+		checkpermission(instance, nil, "bullets.fire")
+		if damage ~= nil then checkluatype(damage, TYPE_NUMBER) damage = math.Clamp(damage, 1, maxBulletDamage:GetInt()) else damage = 1 end
+		if force ~= nil then checkluatype(force, TYPE_NUMBER) force = math.Clamp(force, 0, maxBulletForce:GetInt()) else force = 0 end
+		if distance ~= nil then checkluatype(distance, TYPE_NUMBER) distance = math.Clamp(distance, 0, 32768) else distance = 32768 end
+		if hullSize ~= nil then checkluatype(hullSize, TYPE_NUMBER) hullSize = math.Clamp(hullSize, 0, maxBulletHull:GetInt()) else hullSize = 0 end
+		if num ~= nil then checkluatype(num, TYPE_NUMBER) num = math.Clamp(num, 1, maxBulletNum:GetInt()) else num = 1 end
+
+		local callback
+
+		if cb then
+			checkluatype(cb, TYPE_FUNCTION)
+			callback = function(attacker, tr, dmginfo)
+				instance:runFunction(cb, instance.WrapObject(attacker), SF.StructWrapper(instance, tr, "TraceResult"))
+			end
+		end
+
+		local BulletInfo = {
+			Attacker = instance.player,
+			Callback = callback,
+			Damage = damage,
+			Force = force,
+			Distance = distance,
+			HullSize = hullSize,
+			Num = num,
+			Tracer = 0,
+			TracerName = "",
+			Dir = vunwrap(dir),
+			Spread = vunwrap(spread),
+			Src = vunwrap(src),
+			IgnoreEntity = ignoreEntity and eunwrap(ignoreEntity),
```

Comment:
Move this above as `if ignoreEntity~=nil then ignoreEntity = eunwrap(ignoreEntity) end`